### PR TITLE
test(hx-file-tile): surf-2010

### PR DESCRIPTION
**Pull request by BY00565233**
## Description

unit test `<hx-file-tile>`

image

### What are the relevant story cards/tickets? Any additional PRs or other references?

Jira: SURF-2010 (unit test case)

## Before you request a review for this PR:

- [x] Did you add component tests for any new code?
- [x] Did you run the component unit tests via `yarn test` to ensure all tests passed?
- [x] Did you include a screenshot of the component tests?
- [x] If you changed/added functionality, did you update the demo page and documentation?
- [x] If needed, did you add or modify the demo test page to test the changed/added functionality?
- [x] Did you assign reviewers?
- [x] In Jira, have you linked to this PR on the ticket(s)?

### CODE REVIEW

File: src/elements/hx-file-tile/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+                const shadow = component.shadowRoot.innerHTML;
+
+                expect(component).shadowDom.to.equal(shadow);
+            });
+
+            it('should render the Shadow Root mode open', async () => {
+                const component = /** @type { HXFileTileElement } */ await fixture(template);
+                const mode = component.shadowRoot.mode;
+
+                expect(mode).to.equal("open");
```
Comment (100stacks) [inline src/elements/hx-file-tile/index.spec.js:48]: Please change to single quotes.

File: src/elements/hx-file-tile/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+                expect(queryId).to.equal(id);
+            });
+
+            it('markup should contain a type <hx-icon>', async () => {
+                const elSelector = 'div > a > div > hx-icon';
+                const iconType = 'download';
+                const component = /** @type { HXFileTileElement } */ await fixture(template);
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelector(elSelector);
+                const queryId = query.type;
```
Comment (100stacks) [inline src/elements/hx-file-tile/index.spec.js:103]: Please change variable to `queryType`.

File: src/elements/hx-file-tile/index.spec.js [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+                expect(queryId).to.equal(id);
+            });
+
+            it('markup should contain a #hxRetry button with a times type <button>', async () => {
+                const elSelector = 'div > a > div#hxContentWrapper > div#hxState--invalid > button';
+                const Type = 'button';
+                const component = /** @type { HXFileTileElement } */ await fixture(template);
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelector(elSelector);
+                const queryId = query.type;
```
Comment (100stacks) [inline src/elements/hx-file-tile/index.spec.js:191]: Please change variable to `queryType`.

File: src/elements/hx-file-tile/index.spec.js [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelector(elSelector);
+                const queryId = query.id;
+
+                expect(queryId).to.equal(id);
+            });
+
+            it('markup should contain a #hxRetry button with a times type <button>', async () => {
+                const elSelector = 'div > a > div#hxContentWrapper > div#hxState--invalid > button';
+                const Type = 'button';
```
Comment (100stacks) [inline src/elements/hx-file-tile/index.spec.js:187]: Please change variable to `buttonType`.

File: src/elements/hx-file-tile/index.spec.js [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+                expect(queryId).to.equal(Type);
+            });
+
+            it('markup should contain a type <hx-icon>', async () => {
+                const elSelector = 'div > a > div > div > button > hx-icon';
+                const iconType = 'redo';
+                const component = /** @type { HXFileTileElement } */ await fixture(template);
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelector(elSelector);
+                const queryId = query.type;
```
Comment (100stacks) [inline src/elements/hx-file-tile/index.spec.js:202]: Please change variable to `queryType`.

File: src/elements/hx-file-tile/index.spec.js [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+
+                expect(queryId).to.equal(iconType);
+            });
+
+            it('markup should contain a #hxDismiss button with a times type <button>', async () => {
+                const elSelector = 'div#hxFileTile > button';
+                const Type = 'button';
```
Comment (100stacks) [inline src/elements/hx-file-tile/index.spec.js:209]: Please change variable to `buttonType`.

File: src/elements/hx-file-tile/index.spec.js [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+                const component = /** @type { HXFileTileElement } */ await fixture(template);
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelector(elSelector);
+                const queryId = query.type;
```
Comment (100stacks) [inline src/elements/hx-file-tile/index.spec.js:213]: Please change variable to `queryType`.

File: src/elements/hx-file-tile/index.spec.js [hunk 8]
```diff
@@ -0,0 +1,280 @@
...
+                expect(queryId).to.equal(Type);
+            });
+
+            it('markup should contain a type <hx-icon>', async () => {
+                const elSelector = 'div#hxFileTile > button > hx-icon';
+                const iconType = 'times';
+                const component = /** @type { HXFileTileElement } */ await fixture(template);
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelector(elSelector);
+                const queryId = query.type;
```
Comment (100stacks) [inline src/elements/hx-file-tile/index.spec.js:224]: Please change variable to `queryType`.

File: src/elements/hx-file-tile/index.spec.js (hunk 1 above)
Comment (BY00565233) [reply src/elements/hx-file-tile/index.spec.js]: Changes have been done, Please review

File: src/elements/hx-file-tile/index.spec.js (hunk 2 above)
Comment (BY00565233) [reply src/elements/hx-file-tile/index.spec.js]: Changes have been done, Please review

File: src/elements/hx-file-tile/index.spec.js (hunk 3 above)
Comment (BY00565233) [reply src/elements/hx-file-tile/index.spec.js]: Changes have been done, Please review

File: src/elements/hx-file-tile/index.spec.js (hunk 4 above)
Comment (BY00565233) [reply src/elements/hx-file-tile/index.spec.js]: Changes have been done, Please review

File: src/elements/hx-file-tile/index.spec.js (hunk 5 above)
Comment (BY00565233) [reply src/elements/hx-file-tile/index.spec.js]: Changes have been done, Please review

File: src/elements/hx-file-tile/index.spec.js (hunk 6 above)
Comment (BY00565233) [reply src/elements/hx-file-tile/index.spec.js]: Changes have been done, Please review

File: src/elements/hx-file-tile/index.spec.js (hunk 7 above)
Comment (BY00565233) [reply src/elements/hx-file-tile/index.spec.js]: Changes have been done, Please review

File: src/elements/hx-file-tile/index.spec.js (hunk 8 above)
Comment (BY00565233) [reply src/elements/hx-file-tile/index.spec.js]: Changes have been done, Please review

File: src/elements/hx-file-tile/index.spec.js [hunk 9]
```diff
@@ -0,0 +1,280 @@
...
+            const attr = component.hasAttribute('href');
+            const statusMsg = component.getAttribute('href');
+
+            expect(attr).to.be.true;
+            expect(statusMsg).to.equal(msg);
+        });
+
+        it('should be able to set a progress message', async () => {
+            const component = /** @type {HXFileTileElement} */ await fixture(template);
+            const msg = '50%';
```
Comment (100stacks) [inline src/elements/hx-file-tile/index.spec.js:258]: Please remove the `%` sign.

Comment (BY00565233) [reply src/elements/hx-file-tile/index.spec.js]: Changes have been done, Please review

**Pull request merged**